pkg/receiver/types.go

```diff
@@ -0,0 +1,51 @@
+// Copyright 2020 Comcast Cable Communications Management, LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package receiver
+
+import (
+	"context"
+
+	"github.com/xmidt-org/ears/pkg/event"
+)
+
+//go:generate moq -out testing_mock.go . NewReceiverer Receiver
+
+type InvalidConfigError struct {
+	Err error
+}
+
+type NewReceiverer interface {
+	NewReceiver(config string) (Receiver, error)
+}
+
+type NextFn func(event.Event) error
+
+/*
+Receiver is a plugin that will receive messages and will
+send them to `NextFn`
+
+Other discarded names:
+* Inputter[x]
+* Ingestor[x]
+* Consumer[x]
+* Subscriber[√]
+* Listener[?]  vs Speaker?!?!?
+
+*/
+type Receiver interface {
+	// Hash() string // ReceiverHash ?
+
+	Receive(ctx context.Context, next NextFn) error
```

Comment:
I assume that Receive function is none-blocking? Ie, it will kick off a goroutine to receive events from sources like webhook, kafka, or sqs? If that is the case, I think the Receive function will need to accept an additional errorFn so that it can notify ears core when there is an issue during receiving.

Reply:
Yeah, non-blocking.  Good point on being able to receive errors.  I can imagine with Kafka, there could be a connection issue at some point (or some other failure).  There would need to be an error channel that could be listened to.  Maybe instead of returning `error`, it returns an error channel.  I'll think about this design more.

Reply:
There also needs to be some sort of `Stop()` or `Pause()` type functionality.

Reply:
Related: #40 

`StopReceiving` has been added to the interface.